xarray/core/variable.py

```diff
@@ -1592,19 +1593,39 @@ def _unstack_once(
         else:
             dtype = self.dtype
 
-        data = np.full_like(
-            self.data,
-            fill_value=fill_value,
-            shape=new_shape,
-            dtype=dtype,
-        )
+        if sparse:
+            # TODO: how do we allow different sparse array types
```

Comment:
> TODO: how do we allow different sparse array types

Rather than use a kwarg, could we check the type of what's passed in and, if it's sparse type, then a) use this code path and b) pass a matching type back? i.e. `type(self)(coords=...)`

ref https://github.com/pydata/xarray/pull/4746#issuecomment-753391636

Reply:
It's the other way round. This code unstacks a dense multitindexed array to a sparse array.

Reply:
I got that! But I'm probably underestimating how expensive converting the dense array to sparse would be before passing it in...

Reply:
Oh I see. If the user wants sparse output, they should provide sparse input? That would mean deprecating the `sparse` kwarg right?

Reply:
@dcherian Thanks so much for working on this!   The use case I have is a 1D array of 800,000 points that have at least 4 implicit dimensions.   Converting to a dense array first is not possible; the data are truly sparse, and it would exceed most computer memory limits to convert to dense.   

So instead, I created a Pandas MultiIndex with 800,000 elements that describes the dimensional mapping, and then I call `unstack(sparse=True)` to get a sparse 4-dimensional array.   

If you think this PR addresses my use case, I would like to try out the code but I am not certain what conda environment spec will do this.   Apologies if this is a naive question, but I'm not seeing it covered in the docs.

Another potential concern I have is that I may not be able to save sparse arrays to Zarr format.  I have not tried yet, but I see https://github.com/pydata/sparse/issues/222 and wonder if it's been addressed yet.

Reply:
> I would like to try out the code but I am not certain what conda environment spec will do this

see [Creating a python environment](https://xarray.pydata.org/en/latest/contributing.html#creating-a-python-environment) for instructions

Reply:
@bonnland @max-sixty's proposal is that you convert to a sparse array of 800,000 points before unstacking it to a different sparse array.

Re:writing sparse see https://github.com/pydata/xarray/issues/4156 and https://github.com/pydata/xarray/issues/1077#issuecomment-644803374. There are multiple CF conventions to do this AFAICT (under "discrete sampling geometry").